db/migrate/20230706093050_update_course_progression_limit_from_current_behavior.rb

```diff
@@ -0,0 +1,28 @@
+class UpdateCourseProgressionLimitFromCurrentBehavior < ActiveRecord::Migration[
+  6.1
+]
+  class Course < ApplicationRecord
+    PROGRESSION_BEHAVIOR_LIMITED = -"Limited"
+    PROGRESSION_BEHAVIOR_UNLIMITED = -"Unlimited"
+    PROGRESSION_BEHAVIOR_STRICT = -"Strict"
+  end
+
+  def up
+    Course.all.each do |course|
+      if course.progression_behavior == Course::PROGRESSION_BEHAVIOR_UNLIMITED
+        course.update!(progression_limit: 0)
+      elsif course.progression_behavior == Course::PROGRESSION_BEHAVIOR_LIMITED
+        progression_limit = course.progression_limit + 1
+        course.update!(progression_limit: [progression_limit, 3].min)
```

Comment:
@mahesh-krishnakumar Given that the progression limit is locked to 1, 2 or 3 with the old behaviour, why bother with the call to `.min`? Shouldn't the max possible value of 4 be allowed?

```suggestion
        course.update!(progression_limit: course.progression_limit + 1)
```

Reply:
Yes this was discussed during implementation with @yash-learner . Missed to add it in final today.  Fixed in https://github.com/pupilfirst/pupilfirst/commit/7f87295b27be178ca122d6bac4ca7377b23e776a